modules/blocks/chain.js

```diff
@@ -539,6 +548,17 @@ Chain.prototype.applyBlock = function(block, saveBlock, cb) {
  * @param {boolean} broadcast - Indicator that block needs to be broadcasted
  */
 Chain.prototype.broadcastReducedBlock = function(reducedBlock, broadcast) {
+	if (reducedBlock.totalAmount) {
```

Comment:
Why we need conditions here?

Reply:
because every block might not have `totalAmount`, so checking it, modified the code to 
```
reducedBlock.totalAmount = reducedBlock.totalAmount.toString() || '0';
reducedBlock.totalFee = reducedBlock.totalFee.toString() || '0';
reducedBlock.reward = reducedBlock.reward.toString() || '0';
```

Reply:
That function is only calling the event, why modify block object here before broadcast? That is for sure not the proper place for doing that (it should be when received reduced block or when calling object normalize, etc.).